tools/gce_setup/private_build_and_test.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# This script has to be run from the same directory as grpc_docker.sh and after grpc_docker.sh is sourced
+#
+# Sample Usage:
+# ===============================
+# ./private_build_and_test.sh [language] [environment: interop|cloud] [test case]
+#                              [git base directory] [server name in interop environment] 
+# sh private_build_and_test.sh java interop large_unary /usr/local/google/home/donnadionne/grpc-git grpc-docker-server1
+# sh private_build_and_test.sh java cloud large_unary /usr/local/google/home/donnadionne/grpc-git
+# =============================== 
+
+# Arguments
+LANGUAGE=$1
+ENV=$2
+TEST=$3
+GIT=$4
+SERVER=${5:-"grpc-docker-server"}
+
+sudo docker run --name="private_images" -v $4:/var/local/git-clone grpc/$1 /var/local/git-clone/grpc/tools/dockerfile/grpc_$1/build.sh
+
+sudo docker commit -m "private image" -a "donnadionne" private_images grpc/private_images
+
+sudo docker tag -f grpc/private_images 0.0.0.0:5000/grpc/private_images
+
+sudo docker push 0.0.0.0:5000/grpc/private_images
+
+sudo docker rmi -f grpc/private_images
+
+sudo docker rm private_images
+
+gcloud compute --project "stoked-keyword-656" ssh --zone "asia-east1-a" "grpc-docker-testclients1" --command "sudo docker pull 0.0.0.0:5000/grpc/private_images"
```

Comment:
Let's parameterize the project name to start with at least -- $PROJECT_NAME
perhaps, from a command line arg?

On Wed, Apr 22, 2015 at 12:55 PM donnadionne <notifications@github.com>
wrote:

> In tools/gce_setup/private_build_and_test.sh
> <https://github.com/grpc/grpc/pull/1317#discussion_r28908530>:
>
> > +GIT=$4
> > +SERVER=${5:-"grpc-docker-server"}
> > +
> > +sudo docker run --name="private_images" -v $4:/var/local/git-clone grpc/$1 /var/local/git-clone/grpc/tools/dockerfile/grpc_$1/build.sh
> > +
> > +sudo docker commit -m "private image" -a "donnadionne" private_images grpc/private_images
> > +
> > +sudo docker tag -f grpc/private_images 0.0.0.0:5000/grpc/private_images
> > +
> > +sudo docker push 0.0.0.0:5000/grpc/private_images
> > +
> > +sudo docker rmi -f grpc/private_images
> > +
> > +sudo docker rm private_images
> > +
> > +gcloud compute --project "stoked-keyword-656" ssh --zone "asia-east1-a" "grpc-docker-testclients1" --command "sudo docker pull 0.0.0.0:5000/grpc/private_images"
>
> I see.
>
> No people who are not a part of our project will not be able to use the
> script.
>
> For them the plan is provide a recipe. For example, include instructions
> like they would need a cloud environment where they have vms, some sort of
> a storage system, etc... and then they can take the script, make some
> modifications or pass in some arguments to make it work.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/1317/files#r28908530>.
>